airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -193,6 +193,204 @@ def __init__(self, service, project_id):
+        if compression not in allowed_compressions:
+            raise ValueError("{0} is not a valid compression format. "
+                             "Please use one of the following types: {1}"
+                             .format(compression, allowed_compressions))
+
+        table_resource = {
+            'externalDataConfiguration': {
+                'autodetect': autodetect,
+                'sourceFormat': source_format,
+                'sourceUris': source_uris,
+                'compression': compression,
+                'ignoreUnknownValues': ignore_unknown_values
+            },
+            'tableReference': {
+                'projectId': project_id,
+                'datasetId': dataset_id,
+                'tableId': external_table_id,
+            }
+        }
+
+        if schema_fields:
+            table_resource['externalDataConfiguration'].update({
+                'schema': {
+                    'fields': schema_fields
+                }
+            })
+
+        self.log.info('Creating external table: %s', external_project_dataset_table)
+
+        if max_bad_records:
+            table_resource['externalDataConfiguration']['maxBadRecords'] = max_bad_records
+
+        # if following fields are not specified in src_fmt_configs,
+        # honor the top-level params for backward-compatibility
+        if 'skipLeadingRows' not in src_fmt_configs:
+            src_fmt_configs['skipLeadingRows'] = skip_leading_rows
+        if 'fieldDelimiter' not in src_fmt_configs:
+            src_fmt_configs['fieldDelimiter'] = field_delimiter
+        if quote_character:
+            src_fmt_configs['quote'] = quote_character
+        if allow_quoted_newlines:
+            src_fmt_configs['allowQuotedNewlines'] = allow_quoted_newlines
+        if allow_jagged_rows:
+            src_fmt_configs['allowJaggedRows'] = allow_jagged_rows
+
+        src_fmt_to_param_mapping = {
+            'CSV': 'csvOptions',
+            'GOOGLE_SHEETS': 'googleSheetsOptions',
+            'AVRO': 'AVRO',
+            'NEWLINE_DELIMITED_JSON': 'NEWLINE_DELIMITED_JSON',
+            'DATASTORE_BACKUP': 'DATASTORE_BACKUP'
+        }
+
+        src_fmt_to_configs_mapping = {
+            'csvOptions': [
+                'allowJaggedRows', 'allowQuotedNewlines', 'autodetect',
+                'fieldDelimiter', 'skipLeadingRows', 'ignoreUnknownValues',
+                'nullMarker', 'quote'
+            ],
+            'NEWLINE_DELIMITED_JSON': ['autodetect', 'ignoreUnknownValues'],
+            'AVRO': [],
+            'googleSheetsOptions': ['skipLeadingRows'],
+            'DATASTORE_BACKUP': []
+        }
+
+        valid_configs = src_fmt_to_configs_mapping[
+            src_fmt_to_param_mapping[source_format]
+        ]
+        src_fmt_configs = {
+            k: v
+            for k, v in src_fmt_configs.items() if k in valid_configs
+        }
+        table_resource['externalDataConfiguration'].update(src_fmt_configs)
```

Comment:
I think some of the configs won't take effect the way you'd like them to be.
See skipLeadingRows for example: `externalDataConfiguration.csvOptions.skipLeadingRows`
When it's updated in the table_source, it's updated as `externalDataConfiguration.skipLeadingRows`

Reply:
True. Editing it now.